app/src/main/java/com/krachkovsky/viewpagerdbhomework/ui/MainActivity.kt

```diff
@@ -0,0 +1,86 @@
+package com.krachkovsky.viewpagerdbhomework.ui
+
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.krachkovsky.viewpagerdbhomework.databinding.ActivityMainBinding
+import com.krachkovsky.viewpagerdbhomework.ui.fragment.AddToDatabaseFragment
+import com.krachkovsky.viewpagerdbhomework.ui.fragment.DatabaseListFragment
+
+class MainActivity : FragmentActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        val pagerAdapter = ScreenSlidePagerAdapter(this)
+
+        with(binding) {
+
+            pager.adapter = pagerAdapter
+
+            btnNext.setOnClickListener {
+                if (pager.currentItem < ScreenSlidePagerAdapter(this@MainActivity).itemCount - 1) {
```

Comment:
зачем каждый раз создавать адаптер и получать у него количество элементов?
```suggestion
                if (pager.currentItem < pagerAdapter.itemCount - 1) {
```

Reply:
Исправил